distributed/deploy/cluster.py

```diff
@@ -94,23 +94,56 @@ async def _start(self):
         )
         self._cluster_info.update(info)
 
-        self.periodic_callbacks["sync-cluster-info"] = PeriodicCallback(
-            self._sync_cluster_info, self._sync_interval * 1000
-        )
+        # Start a background task for syncing cluster info with the scheduler
+        self._sync_cluster_info_task = asyncio.ensure_future(self._sync_cluster_info())
```

Comment:
Nit: I think [`asyncio.create_task`](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task) is preferred over `asyncio.ensure_future` for Python 3.7+

```suggestion
        self._sync_cluster_info_task = asyncio.create_task(self._sync_cluster_info())
```